### PR TITLE
AJ-985 Save Columns Modal Accessibility Issues

### PR DESCRIPTION
**Pull request by kevinmarete**
### What
- This PR fixes accessibility issues in the Save Columns feature in the Data tab

### Why
- These changes improve accessibility by resolving critical and serious problems, ensuring the Save Columns feature is usable for all users and meets accessibility standards.

### Testing strategy
- Ran automated accessibility checks with [axe Dev Tools](https://www.deque.com/axe/) to ensure no new issues.

### CODE REVIEW

**Comment by calypsomatic** [comment]
I'm trying to compare your branch v. dev with voiceover and can't figure out the difference.  Do you know what I should be looking for?

**Comment by davidangb** [comment]
I don't see any change in axe devtools on this branch compared to `dev` … and the code changed in this PR is for the autosuggest/input, while AJ-985 highlights the column checkboxes - is this correct?

**Comment by kevinmarete** [comment]
> I'm trying to compare your branch v. dev with voiceover and can't figure out the difference. Do you know what I should be looking for?

Install [axe Dev Tools](https://www.deque.com/axe/) and scan the save columns modal (shown below) with the tool. You should find **0** issues on the modal. 

Screenshot 2024-09-19 at 9 41 32 AM

I expect if you perform a full scan, you can still find issues on the Data Page which will be resolved on [AJ-986](https://broadworkbench.atlassian.net/browse/AJ-986)

Screenshot 2024-09-19 at 9 45 00 AM

[AJ-986]: https://broadworkbench.atlassian.net/browse/AJ-986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kevinmarete** [comment]
> I don't see any change in axe devtools on this branch compared to `dev` … and the code changed in this PR is for the autosuggest/input, while AJ-985 highlights the column checkboxes - is this correct?

The ticket highlighted two issues:
- _Certain ARIA roles must contain particular children 
(This issue was caused when the save selection input tagged with the `role='combobox'` uses an autocomplete suggestions, when is empty or does not contain any suggestions then accessibility requires it to contain children with the `role='option'`)_
- ARIA toggle fields must have an accessible name
_(This issue had been previously resolved in [UIE-101 ](https://github.com/DataBiosphere/terra-ui/pull/4049)with the addition of [`aria-label=''`](https://github.com/DataBiosphere/terra-ui/blob/1b808a758902b7a08d110a185b3d508bb0ba559e/src/components/ColumnSettingsList.ts#L206)

File: src/components/input.js [hunk 1]
```diff
@@ -416,6 +417,7 @@ const withAutocomplete = (WrappedComponent) =>
                         () => [
                           div(
                             {
+                              role: 'option', // Required for screen readers to announce the placeholder text
```
Comment (davidangb) [inline src/components/input.js:420]: nice catch!

File: src/components/input.js [hunk 2]
```diff
@@ -366,6 +366,7 @@ const withAutocomplete = (WrappedComponent) =>
             clearSelectionRef.current = clearSelection;
             return div(
               {
+                'aria-controls': 'autocomplete-suggestions', // Required for screen readers to announce the suggestions container
```
Comment (davidangb) [inline src/components/input.js]: this does not seem right. Is this absolutely necessary? By inspecting the DOM, I can see:
* the downshift library already sets `aria-controls="downshift-0-menu"` on the input element. You're adding an additional `aria-controls` on the div containing the input element.
* the element containing the autosuggestions has an id of `downshift-0-menu`, not `autocomplete-suggestions`

_aria-controls attributes:_
Screenshot 19-09-2024 at 10 22 (2)

_autosuggest options_:
Screenshot 19-09-2024 at 10 25

**Review by calypsomatic — APPROVED** [review]

**Review by davidangb — APPROVED** [review]

File: src/components/input.js (hunk 2 above)
Comment (kevinmarete) [reply src/components/input.js]: As explained, the Downshift component has WrapperComponent that has an attribute `role='listbox'` which requires then `aria-controls=<DownshiftMenuId>`. I have [updated](https://github.com/DataBiosphere/terra-ui/pull/5092/commits/54ca08fc0142289452eece583b3e50757b4a0a42) the **DownshiftMenuId** to reference the dynamic MenuId when opened or the WrapperId when closed.

**Pull request merged**